tests/test_union.py

```diff
@@ -124,7 +123,6 @@ class B:
     class Mutation:
         @strawberry.mutation
         def hello(self, info) -> Union[A, B]:
-            # TODO: mypy is unable to understand that Outside is a dataclass
             return Outside(c=5)  # type:ignore
```

Comment:
Mypy now understands that `Outside` is a dataclass, but then fails since it isn't a member of `Union[A, B]`, which is to be expected.